cpp/src/io/avro/avro_gpu.cu

```diff
@@ -107,22 +142,27 @@ avro_decode_row(schemadesc_s const* schema,
     void* dataptr = schema[i].dataptr;
     switch (kind) {
       case type_null:
-        if (dataptr != nullptr && row < max_rows) {
-          atomicAnd(static_cast<uint32_t*>(dataptr) + (row >> 5), ~(1 << (row & 0x1f)));
+        if (dataptr != nullptr && row_idx >= 0) {
+          atomicAnd(static_cast<uint32_t*>(dataptr) + (row_idx >> 5), ~(1 << (row_idx & 0x1f)));
           atomicAdd(&schema_g[i].count, 1);
+          *skipped_row = false;
```

Comment:
Could we centralize changes to `skipped_row`? Is perhaps `dataptr == nullptr || row_idx < 0` the only condition we need to set `skipped_row` to `true`?